lib/mailchimp3/endpoint.rb

```diff
@@ -34,7 +34,11 @@ def post(body = {})
         body[:apikey] = @oauth_access_token || @basic_auth_key if @version == 2
         req.body = body.to_json
       end
-      _build_response(@last_result)
+      if @last_result.status == 204
```

Comment:
I think the check for 204 should be done down in `_build_response` in the `case`.